pkg/provider/kubernetes/ingress/client.go

```diff
@@ -275,8 +297,53 @@ func extensionsToNetworking(ing marshaler) (*networkingv1beta1.Ingress, error) {
 	return ni, nil
 }
 
+func addServiceFromV1Beta1(ing *networkingv1.Ingress, old networkingv1beta1.Ingress) {
+	if old.Spec.Backend != nil {
+		port := networkingv1.ServiceBackendPort{}
+		if old.Spec.Backend.ServicePort.Type == intstr.Int {
+			port.Number = old.Spec.Backend.ServicePort.IntVal
+		} else {
+			port.Name = old.Spec.Backend.ServicePort.StrVal
+		}
+
+		ing.Spec.DefaultBackend = &networkingv1.IngressBackend{
+			Service: &networkingv1.IngressServiceBackend{
+				Name: old.Spec.Backend.ServiceName,
+				Port: port,
+			},
+		}
+	}
+
+	if ing.Spec.Rules == nil {
+		return
+	}
+
+	for rc, rules := range ing.Spec.Rules {
+		if rules.HTTP == nil || len(rules.HTTP.Paths) == 0 {
+			return
+		}
+		for pc, path := range rules.HTTP.Paths {
+			if path.Backend.Service == nil {
+				port := networkingv1.ServiceBackendPort{}
+				if old.Spec.Rules[rc].HTTP.Paths[pc].Backend.ServicePort.Type == intstr.Int {
```

Comment:
Are we assured that the length of Rules of the old ingress is indentical as the length of the Rules of the networking v1 one ?

Reply:
The same applies to the Paths slice

Reply:
Yes we are :)